File: frontend/src/data/orderService.js
import api from "./api";

// Place Order
export const placeOrder = async (orderData) => {
  const response = await api.post(
    "/orders",
    orderData
  );

  return response.data;
};


// Add Order Item
export const addOrderItem = async (orderItem) => {
  const response = await api.post(
    "/orders/item",
    orderItem
  );

  return response.data;
};


// Get All Orders
export const getAllOrders = async () => {
  const response = await api.get(
    "/orders"
  );

  return response.data;
};


// Get Order By ID
export const getOrderById = async (id) => {
  const response = await api.get(
    `/orders/${id}`
  );

  return response.data;
};


// Get Orders By User
export const getOrdersByUser = async (userId) => {
  const response = await api.get(
    `/orders/user/${userId}`
  );

  return response.data;
};


// Get Order Items
export const getOrderItems = async (orderId) => {
  const response = await api.get(
    `/orders/items/${orderId}`
  );

  return response.data;
};


// Update Order Status
export const updateOrderStatus = async (
  id,
  status
) => {
  const response = await api.put(
    `/orders/status/${id}/${status}`
  );

  return response.data;
};


// Delete Order
export const deleteOrder = async (id) => {
  const response = await api.delete(
    `/orders/${id}`
  );

  return response.data;
};

File: backend/src/main/java/com/drinkit/service/OrderService.java
package com.drinkit.service;

import java.time.LocalDateTime;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.drinkit.entity.Order;
import com.drinkit.entity.OrderItem;
import com.drinkit.repository.OrderItemRepository;
import com.drinkit.repository.OrderRepository;

@Service
public class OrderService {

    @Autowired
    private OrderRepository orderRepository;

    @Autowired
    private OrderItemRepository orderItemRepository;


    // Place Order
    public Order placeOrder(Order order) {

        order.setOrderDate(LocalDateTime.now());

        if (order.getStatus() == null
                || order.getStatus().isEmpty()) {

            order.setStatus("PLACED");
        }

        return orderRepository.save(order);
    }


    // Add Order Item
    public OrderItem addOrderItem(OrderItem orderItem) {

        double total =
                orderItem.getPrice()
                * orderItem.getQuantity();

        orderItem.setTotalPrice(total);

        return orderItemRepository.save(orderItem);
    }


    // Get All Orders
    public List<Order> getAllOrders() {

        return orderRepository.findAll();
    }


    // Get Order By ID
    public Order getOrderById(Long id) {

        return orderRepository
                .findById(id)
                .orElse(null);
    }


    // Get Orders By User
    public List<Order> getOrdersByUser(Long userId) {

        return orderRepository
                .findByUserId(userId);
    }


    // Get Order Items
    public List<OrderItem> getOrderItems(Long orderId) {

        return orderItemRepository
                .findByOrderId(orderId);
    }


    // Update Order Status
    public Order updateStatus(
            Long id,
            String status) {

        Order order =
                orderRepository
                        .findById(id)
                        .orElse(null);

        if (order == null) {
            return null;
        }

        order.setStatus(status);

        return orderRepository.save(order);
    }


    // Delete Order
    public String deleteOrder(Long id) {

        Order order =
                orderRepository
                        .findById(id)
                        .orElse(null);

        if (order == null) {
            return "Order Not Found";
        }

        List<OrderItem> items =
                orderItemRepository
                        .findByOrderId(id);

        orderItemRepository.deleteAll(items);

        orderRepository.delete(order);

        return "Order Deleted Successfully";
    }
}